src/Core/Services/UsernameGenerationService.cs

```diff
@@ -0,0 +1,250 @@
+            {
+                case UsernameType.PlusAddressedEmail:
+                    return await GeneratePlusAddressedEmailAsync(options);
+                case UsernameType.CatchAllEmail:
+                    return await GenerateCatchAllAsync(options);
+                case UsernameType.ForwardedEmailAlias:
+                    return await GenerateForwardedEmailAliasAsync(options);
+                case UsernameType.RandomWord:
+                    return await GenerateRandomWordAsync(options);
+                default:
+                    return string.Empty;
+            }
+        }
+
+        public async Task<UsernameGenerationOptions> GetOptionsAsync()
+        {
+            if (_optionsCache == null)
+            {
+                var options = await _stateService.GetUsernameGenerationOptionsAsync();
+                if (options == null)
+                {
+                    _optionsCache = _defaultOptions;
+                }
+                else
+                {
+                    options.Merge(_defaultOptions);
+                    _optionsCache = options;
+                }
+            }
+
+            return _optionsCache;
+        }
+        public async Task SaveOptionsAsync(UsernameGenerationOptions options)
+        {
+            await _stateService.SetUsernameGenerationOptionsAsync(options);
+            _optionsCache = options;
+        }
+
+        public void ClearCache()
+        {
+            _optionsCache = null;
+        }
+
+        private async Task<string> GenerateRandomWordAsync(UsernameGenerationOptions options)
+        {
+            options.Merge(_defaultOptions);
+            string randomWord = null;
+            if (options.RandomWordUsernameCapitalize == null)
+            {
+                options.RandomWordUsernameCapitalize = false;
+            }
+            if (options.RandomWordUsernameIncludeNumber == null)
+            {
+                options.RandomWordUsernameIncludeNumber = false;
+            }
+            var listLength = EEFLongWordList.Instance.List.Count - 1;
+            var wordIndex = await _cryptoService.RandomNumberAsync(0, listLength);
+            if (options.RandomWordUsernameCapitalize.GetValueOrDefault())
+            {
+                randomWord = Capitalize(EEFLongWordList.Instance.List[wordIndex]);
+            }
+            else
+            {
+                randomWord = EEFLongWordList.Instance.List[wordIndex];
+            }
+
+            if (options.RandomWordUsernameIncludeNumber.GetValueOrDefault())
+            {
+                randomWord = await AppendRandomNumberToRandomWordAsync(randomWord);
+            }
+
+            return randomWord;
+        }
+
+        private async Task<string> GeneratePlusAddressedEmailAsync(UsernameGenerationOptions options)
+        {
+            options.Merge(_defaultOptions);
+            string generatedString = string.Empty;
+            var adressedEmail = options.PlusAddressedEmail;
```

Comment:
you can use directly `options.PlusAddressedEmail` or was it causing problems?

Reply:
Yes, I can, just thought it was easier to read like this. Refactored to use `options.PlusAddressedEmail` .